### PR TITLE
Dev kmr

### PR DESCRIPTION
**Pull request by k-rister**

### CODE REVIEW

**Comment by k-rister** [comment]
Just a note, at the moment this code relies on this `refresh-timestamps` object to be present in `rickshaw-settings.json`:

```
...
    "workshop": {
        "force-builds": "false",
        "refresh-timestamps": {
            "token-file": "/root/quay-oauth.token",
            "api-url": "https://quay.io/api/v1/repository/crucible/client-server"
        }
    },
...
```

File: rickshaw-run [hunk 1]
```diff
@@ -1070,20 +1070,49 @@ sub source_container_image {
...
+                        exit 1;
+                    }
                 } else {
-                    print "\tFailed to refresh expiration timestamp\n";
+                    print "\tFailed to query for tag information\n";
                     exit 1;
                 }
+
+                if ($current_timestamp >= $refresh_timestamp) {
+                    printf "\tThe current expiration timestamp (%d) is greater than or equal to the refresh timestamp so not refreshing\n", $current_timestamp;
```
Comment (rafaelfolco) [inline rickshaw-run:1099]: the wording is confusing, but I don't have any better suggestion

**Review by rafaelfolco — APPROVED** [review]
lgtm

**Comment by k-rister** [comment]
> Just a note, at the moment this code relies on this `refresh-timestamps` object to be present in `rickshaw-settings.json`:
> 
> ```
> ...
>     "workshop": {
>         "force-builds": "false",
>         "refresh-timestamps": {
>             "token-file": "/root/quay-oauth.token",
>             "api-url": "https://quay.io/api/v1/repository/crucible/client-server"
>         }
>     },
> ...
> ```

This has been refactored into this:

```
...
    "quay": {
        "image-expiration": "2w",
        "refresh-expiration": {
            "token-file": "/root/crucible-internal/crucible-production-quay-oauth.token",
            "api-url": "https://quay.io/api/v1/repository/crucible/client-server"
        }
    }
...
```

It just makes more sense for this to be put in the `quay` object and hopefully it's more logical to call it `refresh-expiration` instead of `refresh-timestamps`.

File: rickshaw-settings.json [hunk 2]
```diff
@@ -20,13 +20,13 @@
...
     },
     "workshop": {
-        "force-builds": "false",
-        "refresh-timestamps": {
+        "force-builds": "false"
+    },
+    "quay": {
+        "image-expiration": "2w",
+        "refresh-expiration": {
             "token-file": null,
```
Comment (rafaelfolco) [inline rickshaw-settings.json:28]: if this is just a file path, what is the problem in setting the value ?

File: rickshaw-settings.json [hunk 3]
```diff
@@ -20,7 +20,11 @@
...
     "workshop": {
-      "force-builds": "false"
+        "force-builds": "false",
+        "refresh-timestamps": {
+            "token-file": null,
+            "api-url": null
+        }
     },
     "quay": {
         "image-expiration": "2w"
```
Comment (rafaelfolco) [inline rickshaw-settings.json]: 2w for unused images ? if no runs, it expires in 2w, right ?

**Review by rafaelfolco — APPROVED** [review]
lgtm

File: rickshaw-settings.json (hunk 3 above)
Comment (k-rister) [reply rickshaw-settings.json]: Yes, unless it was built elsewhere (such as CI) with a longer expiration.

File: rickshaw-settings.json (hunk 2 above)
Comment (k-rister) [reply rickshaw-settings.json:28]: Not sure what you mean by this.

**Pull request merged**